Add 'enum_select' diagnostic selection to clang. (#122505)
llvm/llvm-project: clang/docs/InternalsManual.rst

  diagnostic instead of having to do things textually.  The selected string
  does undergo formatting.

**"enum_select format**

Example:
  ``unknown frobbling of a %enum_select<FrobbleKind>{%VarDecl{variable declaration}|%FuncDecl{function declaration}}0 when blarging``
Class:
  Integers
Description:
  This format specifier is used exactly like a ``select`` specifier, except it
  additionally generates a namespace, enumeration, and enumerator list based on
  the format string given. In the above case, a namespace is generated named

>   ``FrobbleKind`` that has a anonymous enumeration with the enumerators

Reviewer: ```suggestion
  ``FrobbleKind`` that has an unscoped enumeration with the enumerators
```
